test/unit/math/test_ad.hpp

```diff
@@ -1526,6 +1629,93 @@ inline void test_matvar_sum_gradient(MatVar&& A_mv, VarMat&& A_vm,
   expect_near_rel("var<Matrix> vs Matrix<var> sum result adjoints",
                   A_vm_f.adj(), A_mv_f.adj(), 1e-12);
 }
+
+/**
+ * Test that the sum of the jacobian for matrices of vars is equal for the
+ *  var matrix.
+ * @tparam MatVar1 An Eigen type inheriting from `EigenBase` that has
+ *  Scalar vars.
+ * @tparam MatVar2 An Eigen type inheriting from `EigenBase` that has
+ *  Scalar vars.
+ * @tparam VarMat1 A `var_value` with an inner type inheriting from `EigenBase`.
+ * @tparam VarMat2 A `var_value` with an inner type inheriting from `EigenBase`.
+ * @tparam ResultMatVar The resulting type of applying a function to `A_mv1`
+ * and `A_mv2`.
+ * @tparam ResultVarMat The result type of applying a function to `A_vm1` and
+ *  `A_vm2`.
+ * @param A_mv1 A eigen type of vars.
+ * @param A_mv2 A eigen type of vars.
+ * @param A_vm1 A eigen type of vars.
+ * @param A_vm2 A eigen type of vars.
+ * @param A_mv_f The result of a function from applying `A_mv1` and `A_mv2`.
+ * @param A_vm_f The result of a function from applying `A_vm1` and `A_vm2`
+ */
+template <typename ResultVarMat, typename ResultMatVar, typename MatVar1,
+          typename MatVar2, typename VarMat1, typename VarMat2>
+inline void test_matvar_sum_gradient(ResultMatVar& A_mv_f, ResultVarMat& A_vm_f,
```

Comment:
I want to delete these sum tests.

I like where they're coming from, however when we test the Jacobian we're testing all the gradients. Testing in one arbitrary direction doesn't give us that much more information.

I think we had this conversation with the original testing framework and went with testing the full Jacobians. In the spirit of merging this will the rest of the test framework eventually, I want to delete these.

Reply:
Yeah I think that's fair. I mostly just wanted to check some arbitary compound expression to make sure adjoints are propagated upwards correctly but if you think the sum version here is not useful I'm fine with removing it.